src/metabase/models/field_values.clj

```diff
@@ -104,13 +104,14 @@
                     {:human-readable-values human-readable-values
                      :status-code           400}))))
 
-(defn- assert-valid-field-values-type
+(defn- assert-valid-type-hash-combo
+  "If type is present, ensure that it is valid, and that a hash_key is provided iff this is an advanced field type."
   [{:keys [type hash_key] :as _field-values}]
-  (when type
+  (let [type (keyword (or type :full))]
```

Comment:
This logic is coupled to line 156. But you don't need this anymore following https://github.com/metabase/metabase/pull/38751/files#r1490613168
```suggestion
```